src/app/modules/products/pages/single-product/single-product.page.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
+import { ProductService } from "@core/products/product.service";
+import { Observable } from "rxjs";
+
+import { Product } from "@shared/models/classes/product.class";
+
+@Component({
+    selector: "app-single-product",
+    templateUrl: "./single-product.page.html",
+    styleUrls: ["./single-product.page.scss"]
+})
+export class SingleProductPageComponent implements OnInit {
+    product$?: Observable<Product | undefined>;
```

Comment:
I see why we need undefined as well. Realistically, when we request data from the server, we may actually redirect to a separate.page if the slug does not exist, so getOne should never return undefined.

Okay to leave as is for now, we will correct later